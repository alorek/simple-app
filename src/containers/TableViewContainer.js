import React, { Component } from 'react';

import Table from '../components/Table';
import TablePagination from '../components/TablePagination';
import SearchBox from '../components/SearchBox';

class TableViewContainer extends Component {

    render() {

        const tableContainerStyles = {
            maxWidth: '1000px',
            margin: '0 auto',
            backgroundColor: '#ffffff'
        };

        return (
            <div className="content-wrapper">
                <div className="table-container card" style={ tableContainerStyles }>
                    <SearchBox />
                    <Table />
                    <TablePagination />
                </div>
            </div>
        );
    }
}

export default TableViewContainer;
