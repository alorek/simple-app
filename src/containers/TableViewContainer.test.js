import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

import TableViewContainer from './TableViewContainer';
import reducer from '../reducers';

test('TableViewContainer renders correctly', () => {

    const store = createStore(reducer);

    const tree = renderer.create(
        <Provider store={ store }>
            <Router>
                <TableViewContainer />
            </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
