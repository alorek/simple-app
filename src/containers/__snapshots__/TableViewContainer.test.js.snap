// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableViewContainer renders correctly 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="table-container card"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "margin": "0 auto",
        "maxWidth": "700px",
      }
    }
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        onKeyUp={[Function]}
        placeholder="Search"
        style={
          Object {
            "background": "transparent",
            "borderRight": "none",
            "boxShadow": "inset 0 1px 0px rgba(0, 0, 0, .075)",
          }
        }
        type="text"
      />
      <span
        className="input-group-addon"
        style={
          Object {
            "background": "transparent",
          }
        }
      >
        <i
          className="glyphicon glyphicon-search"
        />
      </span>
    </div>
    <table
      className="table table-striped"
    >
      <thead>
        <tr>
          <th
            className="sorting-header"
            onClick={[Function]}
          >
            <span
              className="pull-left"
            >
              Resource Type
            </span>
            <span
              className="pull-right glyphicon glyphicon-triangle-top"
            />
          </th>
          <th
            className="sorting-header"
            onClick={[Function]}
          >
            <span
              className="pull-left"
            >
              Display Title
            </span>
          </th>
          <th
            className="sorting-header"
            onClick={[Function]}
          >
            <span
              className="pull-left"
            >
              Media Type
            </span>
          </th>
          <th
            className="sorting-header"
            onClick={[Function]}
          >
            <span
              className="pull-left"
            >
              Language
            </span>
          </th>
          <th
            className="sorting-header"
            onClick={[Function]}
          >
            <span
              className="pull-left"
            >
              
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Agile
          </td>
          <td>
            Agile Development
          </td>
          <td>
            flash
          </td>
          <td>
            en-us
          </td>
          <td>
            <a
              href="/record/2166"
              onClick={[Function]}
            >
              <span
                className="glyphicon glyphicon-eye-open"
              />
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Agile
          </td>
          <td>
            Scrum
          </td>
          <td>
            flash
          </td>
          <td>
            en-us
          </td>
          <td>
            <a
              href="/record/1401"
              onClick={[Function]}
            >
              <span
                className="glyphicon glyphicon-eye-open"
              />
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Agile
          </td>
          <td>
            Test Driven Development
          </td>
          <td>
            pdf
          </td>
          <td>
            en-us
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Programming
          </td>
          <td>
            My Adventures in CSS
          </td>
          <td>
            html
          </td>
          <td>
            en-us
          </td>
          <td>
            <a
              href="/record/2133"
              onClick={[Function]}
            >
              <span
                className="glyphicon glyphicon-eye-open"
              />
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Programming
          </td>
          <td>
            My Adventures in JavaScript
          </td>
          <td>
            flash
          </td>
          <td>
            en-us
          </td>
          <td>
            <a
              href="/record/2189"
              onClick={[Function]}
            >
              <span
                className="glyphicon glyphicon-eye-open"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <ul
      className="pagination pull-right"
    >
      <li
        onClick={[Function]}
      >
        <a>
          «
        </a>
      </li>
      <li
        className="active"
        onClick={[Function]}
      >
        <a>
          1
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          2
        </a>
      </li>
      <li
        onClick={[Function]}
      >
        <a>
          »
        </a>
      </li>
    </ul>
  </div>
</div>
`;
